### ISSUE TITLE
Receiving client error when using `pageExtensions` in next.config.js

### ISSUE THREAD

**Issue by reggie3-braingu** [trpc/trpc#1437]
Hi @KATT, thank you and all the contributors for your hard work with this library.  I am running into the following issue.

**Issue**

After cloning the examples-next-prisma-starter repo, and setting `pageExtensions` in next.config.js, and running `npm run dx` I receive the following error:

![image](https://user-images.githubusercontent.com/90011823/150602799-0b58223a-17e0-4dea-a4d3-3108f3325b90.png)

**trpc version(s):** 

-  "@trpc/client": "^9.17.0",
-  "@trpc/next": "^9.17.0",
-  "@trpc/react": "^9.17.0",
-   "@trpc/server": "^9.17.0"

**How to replicate**
1. clone https://github.com/trpc/examples-next-prisma-starter
2. update the next.config.js file to the following:
```
/**
 * @link https://nextjs.org/docs/api-reference/next.config.js/introduction
 * @type {import('next').NextConfig}
 */
const nextConfig = {
  pageExtensions: ['page.tsx'],
  reactStrictMode: true,
};

module.exports = nextConfig;
```
3. Change the name of `pages/index.tsx` to `pages/index.page.tsx`
4. Start the application by running `npm run dx`
5. Navigate to the application in your browser

Note: I am adding the `.page.` extension to my pages in order to have my .test files adjacent to my page files.  Next attempts to build the test file unless I follow the fix posted here: https://www.reddit.com/r/nextjs/comments/lm0soy/any_way_to_omit_test_files_from_the_build_process/

**Comment by KATT** [comment]
Hey!

Thanks for the sponsoring :)

I don't know why that is happening, could you do an example repo that I can clone and test?

You could try setting `ssr: false` in `_app.tsx` to see if that helps... that causes some hiccups sometimes.

---


@all-contributors add @reggie3-braingu

**Comment by timcole** [comment]
Hey there! Looking at the [docs on next.js](https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions#including-non-page-files-in-the-pages-directory) for this they have a note on there
> **Note**: Make sure you also rename `_document.js`, `_app.js`, `_middleware.js`, as well as files under `pages/api/`.

Just tested it and adding `.page.tsx` to the following files: `src/pages/_app.tsx`, `src/pages/index.tsx`, `src/pages/post/[id].tsx` and adding something like `api.ts` to your `pageExtensions` so you can rename `src/pages/api/trpc/[trpc].ts` to something like `[trpc].api.ts` should solve this for you.

---

Basically this 😃
```
renamed:    src/pages/_app.tsx -> src/pages/_app.page.tsx
renamed:    src/pages/api/trpc/[trpc].ts -> src/pages/api/trpc/[trpc].api.ts
renamed:    src/pages/index.tsx -> src/pages/index.page.tsx
renamed:    src/pages/post/[id].tsx -> src/pages/post/[id].page.tsx
```

```ts
pageExtensions: ['page.tsx', 'api.ts']
```

**Comment by KATT** [comment]
@all-contributors add @reggie3-braingu for financial

**Comment by KATT** [comment]
@all-contributors add @timcole  for mentoring

**Comment by reggie3-braingu** [comment]
@timcole Thanks!  That did the trick.
